core/src/test/java/org/hibernate/ogm/backendtck/type/custom/JpaConvertCustomTypeTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.type.custom;
+
+import java.net.URL;
+import javax.persistence.AttributeConverter;
+
+import org.junit.Test;
+
+import org.hibernate.Session;
+import org.hibernate.ogm.OgmSessionFactory;
+import org.hibernate.ogm.cfg.OgmConfiguration;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.type.descriptor.java.UrlTypeDescriptor;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+/**
+ * Test the JPA @Convert logic in OGM
+ * TODO: should it be in the TCK, that's core code but it's nice to see it tested across all backends
```

Comment:
+1 for having it in the TCK. That way we can e.g. make sure that the two-step conversion works on Redis' hash dialect, which *only* knows of Strings.